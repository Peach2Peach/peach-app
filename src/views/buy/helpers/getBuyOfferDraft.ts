type GetBuyOfferDraftParams = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
}
export const getBuyOfferDraft = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
}: GetBuyOfferDraftParams): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
})
