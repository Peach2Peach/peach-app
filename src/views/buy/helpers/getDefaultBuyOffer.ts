type DefaultOfferProps = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
  kyc?: boolean
}
export const getDefaultBuyOffer = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
  kyc = false,
}: DefaultOfferProps): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
  kyc,
})
