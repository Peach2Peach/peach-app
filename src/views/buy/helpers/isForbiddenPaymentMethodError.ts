import { isPaymentMethod } from '../../../utils/validation/isPaymentMethod'

export const isForbiddenPaymentMethodError = (
  errorMessage: string | null,
  errorDetails: unknown,
): errorDetails is PaymentMethod[] =>
  errorMessage === 'FORBIDDEN' && Array.isArray(errorDetails) && errorDetails.every(isPaymentMethod)
