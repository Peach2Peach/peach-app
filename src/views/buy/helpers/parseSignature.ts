const signaturePattern = /[a-zA-Z0-9/=+]{88}/u

export const parseSignature = (signature: string) => signature.match(signaturePattern)?.pop() || signature
