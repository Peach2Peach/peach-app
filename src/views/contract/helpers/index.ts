export { isPSBTForBatch } from "./isPSBTForBatch";
export { shouldShowTradeStatusInfo } from "./shouldShowTradeStatusInfo";
export { verifyReleasePSBT as verifyPSBT } from "./verifyReleasePSBT";
