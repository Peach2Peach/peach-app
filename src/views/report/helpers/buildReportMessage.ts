import { APPVERSION, BUILDNUMBER, UNIQUEID } from '../../../constants'

type Props = {
  message: string
  shareDeviceID: boolean
  shareLogs: boolean
}

export const buildReportMessage = ({ message, shareDeviceID, shareLogs }: Props) => {
  let messageToSend = message
  if (shareDeviceID) messageToSend += `\n\nDevice ID Hash: ${UNIQUEID}`
  messageToSend += `\n\nApp version: ${APPVERSION} (${BUILDNUMBER})`

  if (shareLogs) messageToSend += '\n\nUser shared app logs, please check crashlytics'

  return messageToSend
}
