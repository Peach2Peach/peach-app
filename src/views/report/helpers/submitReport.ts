import { sendErrors } from '../../../utils/analytics/sendErrors'
import { peachAPI } from '../../../utils/peachAPI'
import { buildReportMessage } from './buildReportMessage'

type Props = {
  email: string
  reason: string
  topic: string
  message: string
  shareDeviceID: boolean
  shareLogs: boolean
}

export const submitReport = ({ email, reason, topic, message, shareDeviceID, shareLogs }: Props) => {
  const messageToSend = buildReportMessage({ message, shareDeviceID, shareLogs })
  if (shareLogs) {
    sendErrors([new Error(`user shared app logs: ${topic} - ${messageToSend}`)])
  }

  return peachAPI.public.contact.sendReport({
    email,
    reason,
    topic,
    message: messageToSend,
  })
}
