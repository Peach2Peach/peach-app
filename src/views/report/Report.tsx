import { useRef } from "react";
import { TextInput } from "react-native";
import { PeachScrollView } from "../../components/PeachScrollView";
import { Screen } from "../../components/Screen";
import { Button } from "../../components/buttons/Button";
import { Checkbox } from "../../components/inputs/Checkbox";
import { EmailInput } from "../../components/inputs/EmailInput";
import { Input } from "../../components/inputs/Input";
import { useSetPopup } from "../../components/popup/Popup";
import { useNavigation } from "../../hooks/useNavigation";
import { useRoute } from "../../hooks/useRoute";
import { useShowErrorBanner } from "../../hooks/useShowErrorBanner";
import { useToggleBoolean } from "../../hooks/useToggleBoolean";
import { useValidatedState } from "../../hooks/useValidatedState";
import { AppPopup } from "../../popups/AppPopup";
import tw from "../../styles/tailwind";
import { useAccountStore } from "../../utils/account/account";
import i18n from "../../utils/i18n";
import { submitReport } from "./helpers/submitReport";

const emailRules = { email: true, required: true };
const required = { required: true };

export const Report = () => {
  const route = useRoute<"report">();
  const navigation = useNavigation();
  const setPopup = useSetPopup();
  const [email, setEmail, isEmailValid, emailErrors] =
    useValidatedState<string>("", emailRules);
  const [topic, setTopic, isTopicValid, topicErrors] = useValidatedState(
    route.params.topic || "",
    required,
  );
  const [message, setMessage, isMessageValid, messageErrors] =
    useValidatedState(route.params.message || "", required);
  const [shareDeviceID, toggleDeviceIDSharing] = useToggleBoolean(
    route.params.shareDeviceID || false,
  );
  const [shareLogs, toggleShareLogs] = useToggleBoolean(false);
  const reason = route.params.reason;
  const publicKey = useAccountStore((state) => state.account.publicKey);

  const showError = useShowErrorBanner();

  const submit = async () => {
    const isFormValid = isEmailValid && isTopicValid && isMessageValid;
    if (!isFormValid) return;

    const { result, error: err } = await submitReport({
      email,
      reason: i18n(`contact.reason.${reason}`),
      topic,
      message,
      shareDeviceID,
      shareLogs,
    });

    if (result) {
      if (publicKey) {
        navigation.navigate("homeScreen", { screen: "settings" });
      } else {
        navigation.navigate("welcome");
      }
      setPopup(<AppPopup id="reportSuccess" />);
      return;
    }

    if (err) showError();
  };

  let $topic = useRef<TextInput>(null).current;
  let $message = useRef<TextInput>(null).current;

  return (
    <Screen header={i18n("contact.title")}>
      <PeachScrollView contentContainerStyle={tw`justify-center grow`}>
        <EmailInput
          onChangeText={setEmail}
          onSubmitEditing={() => $topic?.focus()}
          value={email}
          placeholder={i18n("form.userEmail.placeholder")}
          errorMessage={emailErrors}
        />
        <Input
          onChangeText={setTopic}
          onSubmitEditing={() => $message?.focus()}
          reference={(el) => ($topic = el)}
          value={topic}
          placeholder={i18n("form.topic.placeholder")}
          errorMessage={topicErrors}
        />
        <Input
          style={tw`h-40`}
          onChangeText={setMessage}
          reference={(el) => ($message = el)}
          value={message}
          multiline={true}
          placeholder={i18n("form.message.placeholder")}
          errorMessage={messageErrors}
        />
        {!publicKey && (
          <Checkbox onPress={toggleDeviceIDSharing} checked={shareDeviceID}>
            {i18n("form.includeDeviceIDHash")}
          </Checkbox>
        )}
        <Checkbox onPress={toggleShareLogs} checked={shareLogs}>
          {i18n("form.shareLogs")}
        </Checkbox>
      </PeachScrollView>
      <Button
        style={tw`self-center`}
        onPress={submit}
        disabled={!(isEmailValid && isTopicValid && isMessageValid)}
      >
        {i18n("report.sendReport")}
      </Button>
    </Screen>
  );
};
