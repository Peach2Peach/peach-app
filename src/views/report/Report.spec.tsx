import { render } from "test-utils";
import { setRouteMock } from "../../../tests/unit/helpers/NavigationWrapper";
import { Report } from "./Report";

describe("Report", () => {
  beforeAll(() => {
    setRouteMock({
      name: "report",
      key: "report",
      params: {
        reason: "other",
        topic: "topic",
        message: "message",
      },
    });
  });
  it("should render correctly", () => {
    const { toJSON } = render(<Report />);
    expect(toJSON()).toMatchSnapshot();
  });
});
