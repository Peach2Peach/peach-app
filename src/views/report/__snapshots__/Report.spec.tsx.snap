// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderBottomWidth": 1,
          },
          {
            "paddingTop": 0,
            "zIndex": 1,
          },
          {
            "backgroundColor": "#FFF9F6",
          },
          {
            "borderColor": "#FEEDE5",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
              "paddingBottom": 6,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 6,
            },
            {},
            undefined,
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexBasis": "0%",
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "gap": 4,
            }
          }
        >
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              [
                {
                  "color": "#2B1911",
                  "fontFamily": "Baloo2-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                [
                  {
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 26,
                  },
                  {},
                  {
                    "color": "#2B1911",
                  },
                  {
                    "flexBasis": "0%",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ],
                false,
              ]
            }
          >
            contact Peach
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 10,
              "justifyContent": "flex-end",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          false,
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            },
            {},
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            {
              "flexGrow": 1,
              "justifyContent": "center",
            }
          }
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              onStartShouldSetResponder={[Function]}
            >
              <View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "gap": 4,
                        "justifyContent": "space-between",
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "width": "100%",
                      },
                      {
                        "borderRadius": 12,
                        "borderWidth": 1,
                        "overflow": "hidden",
                      },
                      {
                        "backgroundColor": "#FFFCFA",
                      },
                      {
                        "borderColor": "#2B1911",
                      },
                      false,
                      false,
                      {
                        "marginBottom": 1,
                        "marginTop": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCorrect={false}
                    blurOnSubmit={false}
                    editable={true}
                    keyboardType="email-address"
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="user@email.com"
                    placeholderTextColor="#BAADA7"
                    removeClippedSubviews={false}
                    style={
                      [
                        {
                          "flexShrink": 1,
                          "fontFamily": "Baloo2-Regular",
                          "fontSize": 18,
                          "height": 40,
                          "letterSpacing": 0.18,
                          "lineHeight": 29.25,
                          "paddingBottom": 0,
                          "paddingTop": 0,
                          "width": "100%",
                        },
                        {
                          "color": "#BAADA7",
                        },
                        false,
                        {
                          "borderColor": "transparent",
                          "borderWidth": 1,
                        },
                        false,
                        false,
                      ]
                    }
                    textAlignVertical="center"
                    value=""
                  />
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "gap": 16,
                      }
                    }
                  />
                </View>
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#2B1911",
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 16,
                        "letterSpacing": 0.16,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "fontFamily": "Baloo2-Medium",
                          "fontSize": 14,
                          "letterSpacing": 0.14,
                          "lineHeight": 20,
                          "marginLeft": 12,
                          "marginTop": 4,
                        },
                        {
                          "color": "#DF321F",
                        },
                      ],
                      false,
                    ]
                  }
                >
                   
                </Text>
              </View>
              <View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "gap": 4,
                        "justifyContent": "space-between",
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "width": "100%",
                      },
                      {
                        "borderRadius": 12,
                        "borderWidth": 1,
                        "overflow": "hidden",
                      },
                      {
                        "backgroundColor": "#FFFCFA",
                      },
                      {
                        "borderColor": "#2B1911",
                      },
                      false,
                      false,
                      {
                        "marginBottom": 1,
                        "marginTop": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCorrect={false}
                    blurOnSubmit={false}
                    editable={true}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="topic"
                    placeholderTextColor="#BAADA7"
                    removeClippedSubviews={false}
                    style={
                      [
                        {
                          "flexShrink": 1,
                          "fontFamily": "Baloo2-Regular",
                          "fontSize": 18,
                          "height": 40,
                          "letterSpacing": 0.18,
                          "lineHeight": 29.25,
                          "paddingBottom": 0,
                          "paddingTop": 0,
                          "width": "100%",
                        },
                        {
                          "color": "#2B1911",
                        },
                        false,
                        {
                          "borderColor": "transparent",
                          "borderWidth": 1,
                        },
                        false,
                        false,
                      ]
                    }
                    textAlignVertical="center"
                    value="topic"
                  />
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "gap": 16,
                      }
                    }
                  />
                </View>
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#2B1911",
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 16,
                        "letterSpacing": 0.16,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "fontFamily": "Baloo2-Medium",
                          "fontSize": 14,
                          "letterSpacing": 0.14,
                          "lineHeight": 20,
                          "marginLeft": 12,
                          "marginTop": 4,
                        },
                        {
                          "color": "#DF321F",
                        },
                      ],
                      false,
                    ]
                  }
                >
                   
                </Text>
              </View>
              <View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "gap": 4,
                        "justifyContent": "space-between",
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "width": "100%",
                      },
                      {
                        "borderRadius": 12,
                        "borderWidth": 1,
                        "overflow": "hidden",
                      },
                      {
                        "backgroundColor": "#FFFCFA",
                      },
                      {
                        "borderColor": "#2B1911",
                      },
                      false,
                      false,
                      {
                        "marginBottom": 1,
                        "marginTop": 1,
                      },
                      {
                        "height": 160,
                      },
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCorrect={false}
                    blurOnSubmit={false}
                    editable={true}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="your message..."
                    placeholderTextColor="#BAADA7"
                    removeClippedSubviews={false}
                    style={
                      [
                        {
                          "flexShrink": 1,
                          "fontFamily": "Baloo2-Regular",
                          "fontSize": 18,
                          "height": 40,
                          "letterSpacing": 0.18,
                          "lineHeight": 29.25,
                          "paddingBottom": 0,
                          "paddingTop": 0,
                          "width": "100%",
                        },
                        {
                          "color": "#2B1911",
                        },
                        false,
                        {
                          "borderColor": "transparent",
                          "borderWidth": 1,
                        },
                        {
                          "height": "100%",
                          "justifyContent": "flex-start",
                          "paddingTop": 8,
                        },
                        false,
                      ]
                    }
                    textAlignVertical="top"
                    value="message"
                  />
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "gap": 16,
                      }
                    }
                  />
                </View>
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#2B1911",
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 16,
                        "letterSpacing": 0.16,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "fontFamily": "Baloo2-Medium",
                          "fontSize": 14,
                          "letterSpacing": 0.14,
                          "lineHeight": 20,
                          "marginLeft": 12,
                          "marginTop": 4,
                        },
                        {
                          "color": "#DF321F",
                        },
                      ],
                      false,
                    ]
                  }
                >
                   
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 4,
                    "opacity": 1,
                  }
                }
              >
                <test-file-stub
                  fill="#F56522"
                  style={
                    [
                      {
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#2B1911",
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 16,
                        "letterSpacing": 0.16,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "flexShrink": 1,
                          "fontFamily": "Baloo2-SemiBold",
                          "fontSize": 16,
                          "letterSpacing": 0.16,
                          "lineHeight": 28,
                        },
                        false,
                      ],
                      false,
                    ]
                  }
                >
                  include device ID hash
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 4,
                    "opacity": 1,
                  }
                }
              >
                <test-file-stub
                  fill="#9F8C82"
                  style={
                    [
                      {
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#2B1911",
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 16,
                        "letterSpacing": 0.16,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "flexShrink": 1,
                          "fontFamily": "Baloo2-SemiBold",
                          "fontSize": 16,
                          "letterSpacing": 0.16,
                          "lineHeight": 28,
                        },
                        {
                          "color": "#BAADA7",
                        },
                      ],
                      false,
                    ]
                  }
                >
                  include app logs
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#F56522",
              "borderColor": undefined,
              "borderRadius": 9999,
              "flexDirection": "row",
              "gap": 8,
              "justifyContent": "center",
              "minWidth": 104,
              "opacity": 0.33,
              "paddingBottom": 3,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 3,
            }
          }
        >
          <Text
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              [
                {
                  "color": "#2B1911",
                  "fontFamily": "Baloo2-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                [
                  {
                    "fontFamily": "Baloo2-SemiBold",
                    "fontSize": 13,
                    "letterSpacing": 0.47,
                    "lineHeight": 22,
                    "textTransform": "uppercase",
                  },
                  {},
                  {
                    "color": "#FFFCFA",
                  },
                ],
                false,
              ]
            }
          >
            send report
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
